examples/copy-publish.js

```diff
@@ -0,0 +1,57 @@
+// To run this script, use `node share.js`
+//
+// Uncomment this line to run on an insecure local environment, if you
+// like:
+// process.env.NODE_TLS_REJECT_UNAUTHORIZED = '0';
+
+const permSdk = require('@permanentorg/node-sdk');
+
+// You can pull the "permSession" and "permMFA" cookie values from
+// devtools on any request.  Include
+//
+// 'baseUrl': 'https://local.permanent.org/api'
+//
+// or similar if you want to run this locally or on another
+// environment (by default it will run against prod)
+
+const permanent = new permSdk.Permanent({
+    'sessionToken': 'permSession_COOKIE',
+    'mfaToken': 'permMFA_COOKIE',
+    'archiveNbr': 'YOUR_ARCHIVE',
+});
+
+run();
+
+function randomElement(array) {
+    return array[Math.floor(Math.random() * array.length)];
+}
+
+async function run() {
+    await permanent.init();
+    console.log('session started');
+
+    // I want to copy a random item from "My Files" to "Public"
+    // In other words, we're publishing a random file from "My Files"!
+    const root = await permanent.folder.getMyFilesFolder();
+    const public = await permanent.folder.getPublicFolder();
+    const randomChild = randomElement(root.ChildItemVOs);
+
+    console.log(`Copying "${randomChild.displayName}" to Public...`);
+
+    /*
+    You must call the proper API endpoint to copy a folder vs a record.
+    Right now when you're handed an arbitrary list of ItemVOs like we are doing
+    here, you have to manually check whether an Item is a folder or record.
```

Comment:
You're right that this is a terrible experience for clients of this library. I am also having trouble actually using this in TypeScript, because `ChildItemVOs` is a mixed array of `RecordVO` and `FolderVO`.

```
Property 'uploadFileName' is missing in type 'FolderVO' but required in type 'RecordVO'.
```

Worse, I don't think we expose either type directly, so users can't even directly typecast.

(This is part of why we need to convert these examples to TypeScript - they're easy to write and use in JavaScript, but that is not representative of the actual environments we expect this library to be used in, and allows us to miss type issues.)

Relatedly, you ask a great question

> I'm a bit unclear on the design of the architecture we want on this project... should this be how it's implemented?

It is a faithful representation of a not-great API. I think it makes sense to have these available internally, and there's no harm in exposing them, but I think in practice we want a convenience method that allows copying either. I think that could take a single `FolderVO | RecordVO`, rather than an array, as my data-free suspicion is that coping a single item at a time is the more common use case... and also that makes the types simpler.